src/main.py

```diff
@@ -0,0 +1,202 @@
+    logging.debug(repos_json)
+
+    return repos_json
+
+
+def clean_brackets_content(init_template_comment, end_template_comment, file_name):
+    """
+    Clean the content between the comments given in the filename selected
+    """
+    # Check the lines of the given init_template_comment and end_template_comment variables
+    with open(file_name, "r") as myFile:
+        for num, line in enumerate(myFile):
+            if init_template_comment in line:
+                logging.debug("Init template comment found at line: {}".format(num))
+                init_template_line = num
+            if end_template_comment in line:
+                logging.debug("End template comment found at line: {}".format(num))
+                end_template_line = num
+                break
+    myFile.close()
+
+    # Check if the comment werent specified
+    try:
+        if init_template_line is None or end_template_line is None:
+            exit(1)
+    except UnboundLocalError:
+        logging.error("No INIT_TEMPLATE_COMMENT or END_TEMPLATE_COMMENT were found")
+        exit(1)
+
+    # Read an store all lines into list
+    with open(file_name, "r") as myFile:
+        lines = myFile.readlines()
+    myFile.close()
+
+    # Write in the same file all the lines that weren´t inside the comments specified
+    # This clean all the content between both comments
+    with open(file_name, "w") as myFile:
+        for num, line in enumerate(lines):
+            if num <= init_template_line or num >= end_template_line:
+                myFile.write(line)
+    myFile.close()
+
+
+def write_markdown(init_template_comment, repo_topics, file_name):
+    """
+    Write all the JSON content as a table inside the file_name given
+    """
+
+    # Create a list of lines of the given file_name
+    with open(file_name, "r") as f:
+        in_file = f.readlines()  # in_file is now a list of lines
+
+    # Create a temporal file
+    out_file = []
+    for line in in_file:
+        out_file.append(line)  # Copy each line, one by one
+        if init_template_comment in line:  # Add a new entry after a match
+            out_file.append("Repository|Topics\n")
+            out_file.append("---|---\n")
+            for (
+                repo,
+                topics,
+            ) in repo_topics:  # Generate the table content from the given JSON
+                out_file.append("{}|{}\n".format(repo, ",".join(topics)))
+
+    # Overwrite the original file_name with the new content
+    with open(file_name, "w") as f:
+        f.writelines(out_file)
+
+    logging.info("Markdown write succesfully")
+
+
+if __name__ == "__main__":
+
+    # Setup logger and variables
+    setup_logger()
+    init_template_comment = os.environ.get(
+        "INIT_TEMPLATE_COMMENT", "<!-- BEGIN_DOCS -->"
+    )
+    end_template_comment = os.environ.get("END_TEMPLATE_COMMENT", "<!-- END_DOCS -->")
+    file_name = os.environ.get("FILE_NAME", "README.md")
```

Comment:
I'm thinking about the problem of using always the same file. Wouldn't be better if the name is something descriptive, like Repository_list_{DATE} or something unique to identify and not overwrite accidentally the last file?

Reply:
This script is created to always overwrite the original file because it will be used with other Github Actions to create a correct workflow.

If you execute this Action alone, it won´t do anything because the changes won´t be pushed automatically to the branch.
I will add 2 examples to use this action in a PR workflow and another one to autocommit this changes once we execute this action.